salt/modules/keystoneng.py

```diff
@@ -0,0 +1,382 @@
+import salt.utils
+
+HAS_SHADE = False
+try:
+    import shade
+    from shade.exc import OpenStackCloudException
+    HAS_SHADE = True
+except ImportError:
+    pass
+
+
+def __virtual__():
+    if HAS_SHADE:
+        return 'keystoneng'
```

Comment:
this name should be set in `__virtualname__` and then the `__virtualname__` variable should be returned here.